app/src/main/java/pm/gnosis/heimdall/data/remote/models/push/ServiceMessages.kt

```diff
@@ -43,4 +43,11 @@ sealed class ServiceMessage {
         @Json(name = "v") val v: String,
         @Json(name = "type") val type: String = "rejectTransaction" // Workaround since moshi is not parsing parent or non-constructor fields
     ) : ServiceMessage()
+
+    @JsonClass(generateAdapter = true)
+    data class TypedDataConfirmation(
+        @Json(name = "type") val type: String = "signTypedDataConfirmation", // Workaround since moshi is not parsing parent or non-constructor fields
```

Comment:
I would move type to the last position, so that you can also use unnamed constructors without having to specify it